rosidl_typesupport_opendds_cpp/resource/srv__type_support.cpp.em

```diff
@@ -334,15 +486,23 @@ get_reply_datareader__@(service.namespaced_type.name)(void * untyped_requester)
 void *
 get_request_datareader__@(service.namespaced_type.name)(void * untyped_replier)
 {
-@# TODO: Implement, considering original code in ffe10f9 or earlier
-    return nullptr;
+  using ReplierType = rosidl_typesupport_opendds_cpp::Replier<
+    @(__dds_msg_typesupport_type)RequestWrapper,
+    @(__dds_msg_typesupport_type)ResponseWrapper
+  >;
+  ReplierType * replier = static_cast<ReplierType *>(untyped_replier);
+  return replier->get_request_datareader();
 }
 
 void *
 get_reply_datawriter__@(service.namespaced_type.name)(void * untyped_replier)
```

Comment:
since this function doesn't begin with `rmw_` I'm assuming that we own this interface. If true, then it would be nice to use `shared_ptr` or the OpenDDS equivalent.